@here/harp-datasource-protocol/lib/WorkerTilerProtocol.ts

```diff
@@ -24,7 +25,7 @@ export namespace WorkerTilerProtocol {
     export interface RegisterIndexRequest extends WorkerServiceProtocol.ServiceRequest {
         type: Requests.RegisterIndex;
         id: string;
-        url: string;
+        input: string | GeoJson;
```

Comment:
should this be URL | GeoJson? That at least seems to be used elsewhere.

Reply:
From the worker, the only need is to make the call to the URL, and we can pass the ready url to it as a string instead of the URL class. What would be the benefit of doing it the other way around?

Reply:
Ok, I would do it just for readability, but anyway minor.

Reply:
> would do it just for readability

That wouldn't help either, because this is definition of "message on wire" (where "wire data" is a limited set of Javascript as defined here: https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm).

So, here we can have string or any other representation of URL, like complety decomposed IURLinterface like `interface IURL {
    hash?: string; host?: string; hostname?: string; href?: string ... }` _without_ any methods, expectations about being a objects of actual `URL` class etc.

So, `string` as a common canonical representation of URL/I is good enough and best in this context.